torch/csrc/dynamo/eval_frame.c

```diff
@@ -645,18 +651,15 @@ static PyObject* _custom_eval_frame(
     PyCodeObject* cached_code = (PyCodeObject*)maybe_cached_code;
     // used cached version
     DEBUG_TRACE("cache hit %s", get_frame_name(frame));
+    // Re-enable custom behavior
+    eval_frame_callback_set(callback);
```

Comment:
Assuming we don't need to also do this in the `return NULL` line (L637) because we'd be crashing there.

Reply:
Yes.